### PR TITLE
Add new method `format`

### PR DESCRIPTION
**Pull request by blednaya-luna**
### Checklist for a new method

- [x] Create directory for new method in root in `param-case`
- [x] Place code to `{method-name}/index.js` in CommonJS export in `camelCase` named export
- [x] Add types to `{method-name}/index.d.ts`
- [x] Add tests to `{method-name}/{method-name}.test.ts`
- [x] Add **fork** tests to `{method-name}/{method-name}.fork.test.ts`
- [x] Add **type** tests to `test-typings/{method-name}.ts`
  - Use `// @ts-expect-error` to mark expected type error
  - `import { expectType } from 'tsd'` to check expected return type
- [ ] Add documentation in `{method-name}/reade.md`
  - Add header `Patronum/MethodName`
  - Add section with overloads, if have
  - Add `Motivation`, `Formulae`, `Arguments` and `Return` sections for each overload
  - Add useful examples in `Example` section for each overload
- [ ] Add section to `README.md` in root
  - Add method to table of contents `- [MethodName](#methodname)`
  - Add section `## [MethodName](/method-name 'Documentation')`
  - With summary and simple example

### CODE REVIEW

File: format/format.fork.test.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      "-t0z2df": "Bob",
+      "e0szo1": 30,
+    }
+  `);
+  expect(serialize(secondScope)).toMatchInlineSnapshot(`
+    Object {
+      "-t0z2df": "Kate",
+      "e0szo1": 18,
+    }
+  `);
```
Comment (sergeysova) [inline format/format.fork.test.ts:72]: But you don't test that `_$result` contains actual value

File: format/format.fork.test.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import { fork, serialize, allSettled } from 'effector/fork';
+import { format } from '.';
+
+test('format works in forked scope', async () => {
+  const app = createDomain();
+
+  const changeName = app.createEvent();
+  const $name = app.createStore('Mike').on(changeName, () => 'Bob');
+
+  const _$result = format`My name is ${$name}`;
```
Comment (sergeysova) [inline format/format.fork.test.ts:12]: The same here

File: format/format.fork.test.ts (hunk 1 above)
Comment (sergeysova) [reply format/format.fork.test.ts:72]: You can

```js
expect(scope.getState($result)).
```

File: format/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  b: Store<B>,
+  c: Store<C>,
+  d: Store<D>,
+  e: Store<E>,
+  f: Store<F>,
+  g: Store<G>,
+  h: Store<H>,
+  i: Store<I>,
+  j: Store<J>,
+): Store<string>;
```
Comment (sergeysova) [inline format/index.d.ts:87]:
```suggestion
export function format<T extends Array<Store<any>>>(
  string: TemplateStringsArray,
  ...stores: T,
): Store<string>;
```

[Try it](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgZRtApgGjgYyugQxnVQzgF84AzKCEOAcnSqvRzSgYCguATNgDYF81AK4A7dsAjjq0EEQA8AQTjoAHsXG8AznGVQoBAJ6LS+RQXHGAfHYAUXOHB0wowcQHMAXHAAq6OBCxKjuXjoGRsaYTnAAdAmuGDq+yjEAlL7m6IquYZ42ANw8ODKucARwALy4+EQkHOj2AIzpXKXi5QBG1bWEIY32AERUEBBDbe1l8Pg6ogLwNaNQCjAABgBiY3AAJAgElHtd5Gs8QA)

File: format/format.test.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  changeName();
+  changeAge();
+  expect(argumentHistory(fn)).toMatchInlineSnapshot(`
+    Array [
+      "That Mike is a 28",
+      "That Bob is a 28",
+      "That Bob is a 30",
+    ]
+  `);
+});
```
Comment (sergeysova) [inline format/format.test.ts:47]:
```suggestion
});

test('latest chunk of string is not lost', () => {
  const fn = jest.fn();
  const changeName = createEvent();
  const changeAge = createEvent();

  const $name = createStore('Mike').on(changeName, () => 'Bob');
  const $age = createStore(28).on(changeAge, () => 30);

  const $result = format`That ${$name} is a ${$age} old`;

  $result.watch(fn);
  expect(fn).toHaveBeenCalledWith('That Mike is a 28');

  changeName();
  changeAge();
  expect(argumentHistory(fn)).toMatchInlineSnapshot(`
    Array [
      "That Mike is a 28 old",
      "That Bob is a 28 old",
      "That Bob is a 30 old",
    ]
  `);
});
```

File: index.js [hunk 5]
```diff
@@ -12,3 +12,4 @@ module.exports.splitMap = require('./split-map').splitMap;
 module.exports.spread = require('./spread').spread;
 module.exports.status = require('./status').status;
 module.exports.throttle = require('./throttle').throttle;
+module.exports.format = require('./format').format;
```
Comment (sergeysova) [inline index.js:15]: Don't need to do this by yourself, just run `yarn build:methods`

**Review by sergeysova — CHANGES_REQUESTED** [review]

**Comment by blednaya-luna** [comment]
my english is very bad :( and i don't think what i can fill out documentation correctly

**Review by sergeysova — APPROVED** [review]

File: format/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { Store } from 'effector';
+
+export function format<T extends Array<Store<any>>>(
```
Comment (AlexandrHoroshih) [inline format/index.d.ts:3]: Wouldn't it be better to allow in types not just stores, but primitives too?

Like this:
```ts
const $name = createStore('Bob').on(change, (name) => nameMap.get(name));
const $age = $name.map(name => ageMap.get(name));

const formatted = format`${window.BASE_STRING} ${$name} ${$age}`;
```
It works correctly without typescript:
https://share.effector.dev/EKLxy7A1
so why not support this at types level

Types for that can be [borrowed from `forest`'s types](https://github.com/effector/effector/blob/261007707c52ddd6995fc80e4042e6348a05c58c/src/forest/index.h.ts#L50) 

 Everything else looks cool, nice work

**Review by sergeysova — APPROVED** [review]

**Comment by sergeysova** [comment]
Thank you!

**Pull request merged**